### PR TITLE
CATROID-802 Refactor StoreCSVIntoUserListAction to Kotlin

### PR DESCRIPTION
**Pull request by qlin93**
Refactored StoreCSVIntoUserListAction to Kotlin

https://jira.catrob.at/browse/CATROID-802

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+            val allRows = reader.readAll()
+            if (allRows.size > 0 && columnToExtract >= 0) {
+                for (row in allRows) {
+                    if (row.size <= columnToExtract) {
+                        userList!!.addListItem("")
+                    } else {
+                        resetList = false
+                        userList!!.addListItem(row[columnToExtract])
+                    }
```
Comment (Koell) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt:84]: please change in line 80&83 the userList!! to userList?
if you don't check for NPE don't enforce them, with the elvis-operator(?) kotlin ignores the function when it is NULL

**Review by Koell — CHANGES_REQUESTED** [review]

File: catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt (hunk 1 above)
Comment (Koell) [reply catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt:84]: the rest of the conversion was fine

**Review by Koell — APPROVED** [review]
:+1:

**Review by gPathpp — CHANGES_REQUESTED** [review]
Well done!
Please, do not only rewrite the code to Kotlin but also try to improve the code quality (Clean Code).
Thanks!

File: catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    fun setFormulaCSVData(formulaCSVData: Formula) {
+        this.formulaCSVData = formulaCSVData
+    }
+
+    fun setSprite(sprite: Sprite) {
+        this.sprite = sprite
+    }
+}
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt:145]: Well done!

PLEASE EXPLAIN: Does the changed behaviour affect the program? (After any Exception you return a default value and also the UserList reset is missing)
If this is not intended, I would suggest returning null and using "?: return" in the update method.

PLEASE ADD (if missing): a test for its original behaviour.

**Review by gPathpp — CHANGES_REQUESTED** [review]

**Comment by qlin93** [comment]
@gPathpp Thanks! The changes did not affect the program. By returning default values the update method would return without making any changes to the UserList, therefore the reset was unnecessary. But since your suggestion with returning null and using "?:return" is a lot cleaner and easier to read I have changed it.  I also made the member variables public. 

Regarding the test, there are five existing Catrobat language tests that test the intended behaviour of the program.

**Review by gPathpp — APPROVED** [review]
LGTM!

**Comment by wslany** [comment]
Thanks!

**Pull request merged**